resources/assets/lib/notification-icon.tsx

```diff
@@ -1,53 +1,39 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-import { computed } from 'mobx';
 import { observer } from 'mobx-react';
-import { typeNames } from 'models/notification-type';
 import core from 'osu-core-singleton';
 import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
 
 interface Props {
+  count: number;
   type?: string;
 }
 
 @observer
 export default class NotificationIcon extends React.Component<Props> {
-  @computed
-  private get unreadCount() {
-    // TODO: need a better way of propagating the exclusion list to this (but it's global anyway?)
-    const types = typeNames.filter((name) => !(name == null || name === 'channel'));
-    return types.reduce((acc, current) => acc + core.dataStore.notificationStore.unreadStacks.getOrCreateType({ objectType: current }).total, 0);
-  }
-
   render() {
+    const modifiers = {
+      glow: this.props.count > 0,
+      mobile: this.props.type === 'mobile',
+    };
+
     return (
-      <span className={this.mainClass()}>
-        <i className='fas fa-inbox' />
+      <span className={classWithModifiers('notification-icon', modifiers)}>
+        <i className='fas fa-comment-alt' />
         <span className='notification-icon__count'>
           {this.unreadCountDisplay()}
         </span>
       </span>
     );
   }
 
-  private mainClass() {
-    let ret = 'notification-icon';
-
-    if (this.unreadCount > 0) {
-      ret += ' notification-icon--glow';
-    }
-
-    if (this.props.type === 'mobile') {
-      ret += ' notification-icon--mobile';
-    }
-
-    return ret;
-  }
-
   private unreadCountDisplay() {
     if (core.notificationsWorker.hasData) {
```

Comment:
I wonder if this should be moved to the chat/main themselves and make this component non-observer

or the other way where the parent component just passing relevant function and this component observes all the changes...